tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Common/MatterProvision.cs

```diff
@@ -1147,35 +1177,33 @@ private static string TeamMembersPermissionInformation(MatterDetails matterDetai
 
             List<string> keys = new List<string>();
             Dictionary<string, string> propertyList = new Dictionary<string, string>();
-            keys.Add(matterSettings.StampedPropertyPracticeGroup);
-            keys.Add(matterSettings.StampedPropertyAreaOfLaw);
-            keys.Add(matterSettings.StampedPropertySubAreaOfLaw);
-            keys.Add(matterSettings.StampedPropertyMatterName);
-            keys.Add(matterSettings.StampedPropertyMatterID);
-            keys.Add(matterSettings.StampedPropertyClientName);
-            keys.Add(matterSettings.StampedPropertyClientID);
-            keys.Add(matterSettings.StampedPropertyResponsibleAttorney);
-            keys.Add(matterSettings.StampedPropertyTeamMembers);
-            keys.Add(matterSettings.StampedPropertyIsMatter);
-            keys.Add(matterSettings.StampedPropertyOpenDate);
-            keys.Add(matterSettings.StampedPropertySecureMatter);
-            keys.Add(matterSettings.StampedPropertyBlockedUploadUsers);
-            keys.Add(matterSettings.StampedPropertyMatterDescription);
-            keys.Add(matterSettings.StampedPropertyConflictCheckDate);
-            keys.Add(matterSettings.StampedPropertyConflictCheckBy);
-            keys.Add(matterSettings.StampedPropertyMatterCenterRoles);
-            keys.Add(matterSettings.StampedPropertyMatterCenterPermissions);
-            keys.Add(matterSettings.StampedPropertyMatterCenterUsers);
-            keys.Add(matterSettings.StampedPropertyDefaultContentType);
-            keys.Add(matterSettings.StampedPropertyIsConflictIdentified);
-            keys.Add(matterSettings.StampedPropertyDocumentTemplateCount);
-            keys.Add(matterSettings.StampedPropertyBlockedUsers);
-            keys.Add(matterSettings.StampedPropertyMatterGUID);
-            keys.Add(matterSettings.StampedPropertyMatterCenterUserEmails);
-
-            propertyList.Add(matterSettings.StampedPropertyPracticeGroup, WebUtility.HtmlEncode(matterDetails.PracticeGroup));
-            propertyList.Add(matterSettings.StampedPropertyAreaOfLaw, WebUtility.HtmlEncode(matterDetails.AreaOfLaw));
-            propertyList.Add(matterSettings.StampedPropertySubAreaOfLaw, WebUtility.HtmlEncode(matterDetails.SubareaOfLaw));
+            //Get all the matter stamped properties from the appsettings.json file
+            var matterStampedProperties = configuration.GetSection("Matter").GetChildren();

```

Comment:
In the new implementation, all the matter property bag keys must start with "stampedproperty". So if in the future any property has been added it will read from with in the loop and no need to make any changes to the code